### PR TITLE
Enhance part span

### PR DESCRIPTION
**Pull request by vrajmohan**

### CODE REVIEW

File: regparser/history/annual.py [hunk 1]
```diff
@@ -20,6 +21,17 @@
...
 
+# Matches any of the following:
+#    Parts 200 to 219
+#    Parts 200 to end
+#    Part 52 (§§ 52.1019 to 52.2019)
+PART_SPAN_REGEX = re.compile(
+    r'.*parts? ('
+    r'(?P<span>(?P<start>\d+) to ((?P<end>\d+)|(?P<end_literal>end)).*)'
+    r'|((?P<single_part>\d+) \(.*\))'
+    r')',
```
Comment (cmc333333) [inline regparser/history/annual.py]: Do we need the outermost parens (lines 29 and 32)? Might read ever so slightly better without them.

Also, would it be worthwhile to convert this to pyparsing? I find parser combinators to be much easier to reason about than regexes.

File: regparser/history/annual.py [hunk 2]
```diff
@@ -49,15 +61,18 @@ def part_span(self):
                     part_string = line
                     break
             if part_string:
-                match = re.match(r'.*parts? (\d+) to (\d+|end).*',
-                                 part_string.lower())
+                match = PART_SPAN_REGEX.match(part_string)
                 if match:
-                    start = int(match.group(1))
```
Comment (cmc333333) [inline regparser/history/annual.py]: To make this a wee bit less "pointy", would it make sense to have:
```python
if match and match.group('span'):
    ...
elif match:
    ...
```

**Comment by cmc333333** [comment]
Good work; I've commented on some very minor style things. Take 'em or leave 'em, but let us know so we know when to merge.

File: regparser/history/annual.py (hunk 1 above)
Comment (vrajmohan) [inline regparser/history/annual.py]: I tried removing it, in vain. For some reason, it doesn't work in Python. Oddly, it works here:
http://pythex.org/?regex=.*parts%3F%20(%3FP%3Cspan%3E(%3FP%3Cstart%3E%5Cd%2B)%20to%20((%3FP%3Cend%3E%5Cd%2B)%7C(%3FP%3Cend_literal%3Eend)).*)%7C((%3FP%3Csingle_part%3E%5Cd%2B)%20%5C(.*%5C))&test_string=Parts%2045%20(foo)%0A&ignorecase=1&multiline=0&dotall=0&verbose=0

**Comment by cmc333333** [comment]
:+1:

**Pull request merged**